python/src/iceberg/types.py

```diff
@@ -126,6 +138,10 @@ def precision(self) -> int:
     def scale(self) -> int:
         return self._scale
 
+    def truncate(self, value: Decimal, width: int) -> Decimal:
+        """Truncate a given decimal value into a given width."""
+        return transform_util.truncate_decimal(value, width)
```

Comment:
Should this be a static method since it doesn't use the instance? I also think the function is only a handful of lines so instead of importing it from `transform_util` should we just add the logic here?
```py
    @staticmethod
    def truncate(value: Decimal, width: int) -> Decimal:
        """Truncate a given decimal value into a given width."""
        value_tuple = value.as_tuple()
        unscaled_value = int(("-" if value_tuple.sign else "") + "".join([str(d) for d in value_tuple.digits]))
        applied_value = unscaled_value - (((unscaled_value % width) + width) % width)
        return Decimal(f"{applied_value}e{value.as_tuple().exponent}")
```